src/NCronJob/Registry/JobRegistry.cs

```diff
@@ -19,7 +21,7 @@ internal sealed class JobRegistry
     public JobDefinition GetJobDefinition<T>() => allJobs.First(j => j.Type == typeof(T));
 
     public JobDefinition? FindJobDefinition(Type type)
```

Comment:
I think we can remove the "TODO" itself. In its current state we only have a jobdefinition.

Reply:
In that context, how about replacing `FirstOrDefault()` by `SingleOrDefault()`?

This would enforce the expected intent of the design or would allow to not swallow potential issues.

Reply:
I am a big fan of "fail fast". Do we have an option for `v3` to throw an exception when we add a duplicate? 
In v4 `UseNCronJob` could be our savior - but not sure what can be done for v3.

If there isn't anything, maybe a custom exception might be more appropriate. Not sure what a user can do if he receives the `InvalidOperationException` with "Sequence contains more than one element" message.

Reply:
I've eventually only removed the TODO.

The code actually doesn't allow duplicates. In that context, Using `FirstOrDefault()` or `SingleOrDefault()` is similar.

However, the way the code deals with duplicates isn't uniform, from a user standpoint. A such, I've created #128 (which I think should be dealt with in the context of v4 as it will certainly generate a change the behavior).